### PR TITLE
Using buildsrc and javapackager 

### PR DESCRIPTION
**Pull request by rgozim**
This PR, essentially, makes insight _distributable_ and cleans up logic from the build.gradle file.

### Test 1
Test the application be run from gradle as Inisght and Importer

1. `gradle run` - runs insight
2. `gradle runImporter` - runs importer

### Test 2
Test an ImageJ jar is built 

1. `gradle imageJJar` creates a jar and outputs it to directory `build/libs`
2  Unzip the jar and check the _MANIFEST.MF_ (inside _META-INF_ folder) contains main-class org.openmicroscopy.shoola.MainIJPlugin

Edit: J-M
1. Unzip ``omero-insight-imagej-5.5.0-SNAPSHOT.zip`` under ``build/distributions`` and move the folder in the plugins folder of Fiji. Check that you can connect.

### Test 3
Create an installer using javapackager

1. ensure you are using Oracle JDK 1.8 and $JAVA_HOME path
      - run `javapackager` in cmd/terminal, it will fail if something isn't right
2. `gradle packageApplicationDmg` (if on Mac) or `gradle packageApplicationExe` if on Windows

Extras:
Windows openjdk with javapackager https://github.com/ojdkbuild/ojdkbuild/releases/download/1.8.0.201-1/java-1.8.0-openjdk-1.8.0.201-1.b09.ojdkbuild.windows.x86_64.msi

### CODE REVIEW

File: buildSrc/build.gradle [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+}
+
+repositories {
+    mavenLocal()
+    jcenter()
+}
+
+dependencies {
+    implementation "commons-io:commons-io:2.6"
+    implementation "org.openmicroscopy:omero-javapackager-plugin:5.5.0-SNAPSHOT"
```
Comment (jburel) [inline buildSrc/build.gradle]: needs to point to m1

File: buildSrc/build.gradle [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+}
+
+gradlePlugin {
+    plugins {
+        insight {
+            id = "org.openmicroscopy.insight"
+            implementationClass = "org.openmicroscopy.InsightPlugin"
+        }
+    }
+}
```
Comment (jburel) [inline buildSrc/build.gradle]: Add extra line

File: settings.gradle [hunk 3]
```diff
@@ -2,7 +2,7 @@ pluginManagement {
     repositories {
         mavenLocal()
         gradlePluginPortal()
-        maven { url 'https://artifacts.openmicroscopy.org/artifactory/maven/' }
+        jcenter()
```
Comment (jburel) [inline settings.gradle]: this is not going to work, our jars won't be found

File: buildSrc/src/main/groovy/org/openmicroscopy/InsightPlugin.groovy [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+            @Override
+            void execute(CreateStartScripts startScripts) {
+                startScripts.setDescription("Creates OS specific scripts to run the project as a JVM application.");
+                startScripts.setClasspath(project.tasks.getByName(JavaPlugin.JAR_TASK_NAME).outputs.files +
+                        project.configurations.getByName(JavaPlugin.RUNTIME_CLASSPATH_CONFIGURATION_NAME))
+                startScripts.mainClassName = MAIN_INSIGHT
+                startScripts.applicationName = "OMERO.importer"
+                startScripts.outputDir = new File(project.buildDir, "scripts")
+                startScripts.executableDir = "bin"
+                startScripts.defaultJvmOpts = ["-Xms256m", "-Xmx1024m"]
```
Comment (jburel) [inline buildSrc/src/main/groovy/org/openmicroscopy/InsightPlugin.groovy]: use DEFAULT_JVM_ARGS

File: buildSrc/src/main/groovy/org/openmicroscopy/InsightPlugin.groovy [hunk 5]
```diff
@@ -0,0 +1,262 @@
...
+
+    private TaskProvider<CreateStartScripts> addInsightStartScript() {
+        project.tasks.register(TASK_IMPORTER_START_SCRIPTS, CreateStartScripts, new Action<CreateStartScripts>() {
+            @Override
+            void execute(CreateStartScripts startScripts) {
+                startScripts.setDescription("Creates OS specific scripts to run the project as a JVM application.");
+                startScripts.setClasspath(project.tasks.getByName(JavaPlugin.JAR_TASK_NAME).outputs.files +
+                        project.configurations.getByName(JavaPlugin.RUNTIME_CLASSPATH_CONFIGURATION_NAME))
+                startScripts.mainClassName = MAIN_INSIGHT
+                startScripts.applicationName = "OMERO.importer"
```
Comment (jburel) [inline buildSrc/src/main/groovy/org/openmicroscopy/InsightPlugin.groovy]: declare static field

File: buildSrc/src/main/groovy/org/openmicroscopy/InsightPlugin.groovy [hunk 6]
```diff
@@ -0,0 +1,262 @@
...
+            @Override
+            void execute(Jar jar) {
+                jar.manifest.attributes(createBasicManifest(mainClass))
+                jar.manifest.attributes["Class-Path"] =
+                        getRuntimeClasspathConfiguration().collect { it.name }.join(" ")
+            }
+        }
+    }
+
+    private Map<String, ?> createBasicManifest(String mainClass) {
```
Comment (jburel) [inline buildSrc/src/main/groovy/org/openmicroscopy/InsightPlugin.groovy]: should we use the existing plugin to create the manifest?

Comment (rgozim) [reply buildSrc/src/main/groovy/org/openmicroscopy/InsightPlugin.groovy]: Insight doesn't use the [publishing plugin](https://github.com/ome/omero-artifact-plugin/blob/e7551f9bb3bded3adc53f279bf97f0cbbc97f701/src/main/kotlin/org/openmicroscopy/PublishingPlugin.kt#L34) which creates that manifest

Comment (jburel) [reply buildSrc/src/main/groovy/org/openmicroscopy/InsightPlugin.groovy]: maybe it could

Comment (rgozim) [reply buildSrc/src/main/groovy/org/openmicroscopy/InsightPlugin.groovy]: Guess it depends on if insight needs to publish a jar somewhere?

Comment (jburel) [reply buildSrc/src/main/groovy/org/openmicroscopy/InsightPlugin.groovy]: I can see that being the case for imagej-jar

File: buildSrc/build.gradle [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+
+repositories {
+    mavenLocal()
+    jcenter()
+    maven { url 'https://artifacts.openmicroscopy.org/artifactory/maven/' }
+}
+
+dependencies {
+    implementation "commons-io:commons-io:2.6"
```
Comment (jburel) [inline buildSrc/build.gradle]: surprise that we need it since it is already a dependency of omero-romio

**Comment by jburel** [comment]
Also no longer having a publishing option fails the build system https://web-proxy.openmicroscopy.org/west-ci/job/OMERO-insight-build/

**Comment by jburel** [comment]
@rgozim can you please do not force pushed you have removed my commits fixing the build?

**Comment by jburel** [comment]
* tar are created. We won't need them
* 2 zips with same content are created OMERO.insight and omero-insight. We only need one

**Comment by manics** [comment]
Building a DMG fails for me, does it require additional dependencies?
```
$ javac -version
javac 1.8.0_112

$ javapackager -version
Java Packager version 8.0

$ JAVA_HOME=/Library/Java/JavaVirtualMachines/jdk1.8.0_112.jdk/Contents/Home gradle packageApplicationDmg
Starting a Gradle Daemon, 10 stopped Daemons could not be reused, use --status for details

> Task :packageApplicationDmg
No base JDK. Package will use system JRE.
No base JDK. Package will use system JRE.
Building DMG package for OMERO.insight
Preparing Info.plist: /var/folders/z0/tv7cs0n16m54ytpp_2tsk2c9w3bcxl/T/fxbundler698124973197547335/macosx/Info.plist
  Using default package resource [Bundle config file]  (add package/macosx/Info.plist to the class path to customize)
   Using custom package resource [icon]  (loaded from file /private/tmp/omero-insight/icons/omeroinsight.icns)
  Config files are saved to /var/folders/z0/tv7cs0n16m54ytpp_2tsk2c9w3bcxl/T/fxbundler698124973197547335/macosx. Use them to customize package.
  Using default package resource [dmg background]  (add package/macosx/OMERO.insight-background.png to the class path to customize)
   Using custom package resource [volume icon]  (loaded from file /private/tmp/omero-insight/icons/omeroinsight.icns)
Using default package resource [script to run after application image is populated]  (add package/macosx/OMERO.insight-post-image.sh to the class path to customize)
Preparing dmg setup: /var/folders/z0/tv7cs0n16m54ytpp_2tsk2c9w3bcxl/T/fxbundler698124973197547335/macosx/OMERO.insight-dmg-setup.scpt
  Using default package resource [DMG setup script]  (add package/macosx/OMERO.insight-dmg-setup.scpt to the class path to customize)
 Creating DMG file: /private/tmp/omero-insight/build/packaged/main/bundles/OMERO.insight-1.0.dmg
Running [/usr/bin/hdiutil, create, -quiet, -srcfolder, /var/folders/z0/tv7cs0n16m54ytpp_2tsk2c9w3bcxl/T/fxbundler698124973197547335/images/image-3755657565764710039, -volname, OMERO.insight, -ov, /var/folders/z0/tv7cs0n16m54ytpp_2tsk2c9w3bcxl/T/fxbundler698124973197547335/images/OMERO.insight-tmp.dmg, -format, UDRW]
Running [/usr/bin/hdiutil, attach, /var/folders/z0/tv7cs0n16m54ytpp_2tsk2c9w3bcxl/T/fxbundler698124973197547335/images/OMERO.insight-tmp.dmg, -quiet, -mountroot, /var/folders/z0/tv7cs0n16m54ytpp_2tsk2c9w3bcxl/T/fxbundler698124973197547335/images]
Running [/usr/bin/SetFile, -c, icnC, /var/folders/z0/tv7cs0n16m54ytpp_2tsk2c9w3bcxl/T/fxbundler698124973197547335/images/OMERO.insight/.VolumeIcon.icns]
Running [/usr/bin/SetFile, -a, C, /var/folders/z0/tv7cs0n16m54ytpp_2tsk2c9w3bcxl/T/fxbundler698124973197547335/images/OMERO.insight]
Running [/usr/bin/hdiutil, detach, -quiet, /var/folders/z0/tv7cs0n16m54ytpp_2tsk2c9w3bcxl/T/fxbundler698124973197547335/images/OMERO.insight]
Running [/usr/bin/hdiutil, convert, /var/folders/z0/tv7cs0n16m54ytpp_2tsk2c9w3bcxl/T/fxbundler698124973197547335/images/OMERO.insight-tmp.dmg, -quiet, -format, UDZO, -o, /private/tmp/omero-insight/build/packaged/main/bundles/OMERO.insight-1.0.dmg]
java.io.IOException: Exec failed with code 1 command [[/usr/bin/hdiutil, convert, /var/folders/z0/tv7cs0n16m54ytpp_2tsk2c9w3bcxl/T/fxbundler698124973197547335/images/OMERO.insight-tmp.dmg, -quiet, -format, UDZO, -o, /private/tmp/omero-insight/build/packaged/main/bundles/OMERO.insight-1.0.dmg] in unspecified directory
        at com.oracle.tools.packager.IOUtils.exec(IOUtils.java:165)
        at com.oracle.tools.packager.IOUtils.exec(IOUtils.java:138)
        at com.oracle.tools.packager.IOUtils.exec(IOUtils.java:132)
        at com.oracle.tools.packager.mac.MacDmgBundler.buildDMG(MacDmgBundler.java:419)
        at com.oracle.tools.packager.mac.MacDmgBundler.bundle(MacDmgBundler.java:92)
        at com.oracle.tools.packager.mac.MacDmgBundler.execute(MacDmgBundler.java:549)
        at com.sun.javafx.tools.packager.PackagerLib.generateNativeBundles(PackagerLib.java:352)
        at com.sun.javafx.tools.packager.PackagerLib.generateDeploymentPackages(PackagerLib.java:319)
        at com.sun.javafx.tools.packager.Main.main(Main.java:476)
  Config files are saved to /var/folders/z0/tv7cs0n16m54ytpp_2tsk2c9w3bcxl/T/fxbundler698124973197547335/macosx. Use them to customize package.
Exception in thread "main" com.sun.javafx.tools.packager.PackagerException: Error: Bundler "DMG Installer" (dmg) failed to produce a bundle.
        at com.sun.javafx.tools.packager.PackagerLib.generateNativeBundles(PackagerLib.java:354)
        at com.sun.javafx.tools.packager.PackagerLib.generateDeploymentPackages(PackagerLib.java:319)
        at com.sun.javafx.tools.packager.Main.main(Main.java:476)

> Task :packageApplicationDmg FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':packageApplicationDmg'.
> Process 'command 'javapackager'' finished with non-zero exit value 1

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 15s
7 actionable tasks: 1 executed, 6 up-to-date
```

**Comment by dominikl** [comment]
Tested on OSX: Same here, javapackager fails (`Unable to locate an executable at "/Library/Java/JavaVirtualMachines/adoptopenjdk-8.jdk/Contents/Home/bin/javapackager"`). The other builds work, but there is no `omero-insight-imagej-5.5.0-SNAPSHOT.zip` just the jar. Copying that into Fiji's plugin directory apparently isn't enough, plugin is not available then.

**Comment by rgozim** [comment]
Ensure you have an oracle version of Java installed on your system. 

For example on osx: 
`brew tap caskroom/versions`
`brew cask install java8`

**Comment by mtbc** [comment]
https://github.com/openmicroscopy/bioformats/pull/3286#discussion_r240237070 raises some Oracle concerns of which we should probably be mindful if using this approach.

**Comment by jburel** [comment]
@dominikl the imageJ zip is under ``build/distributions``

**Comment by jburel** [comment]
version of the .exe is ``1.0`` i.e. the version is not correctly set

**Comment by jburel** [comment]
java.lang.Exception: Abnormal termination due to an uncaught exception.
java.lang.NoSuchMethodError: com.sun.jna.Native.load(Ljava/lang/String;Ljava/lang/Class;Ljava/util/Map;)Lcom/sun/jna/Library;
	at com.sun.jna.platform.win32.Version.<clinit>(Version.java:36)
	at org.openmicroscopy.shoola.env.data.model.appdata.WindowsApplicationDataExtractor.extractAppData(WindowsApplicationDataExtractor.java:235)
	at org.openmicroscopy.shoola.env.data.model.ApplicationData.<init>(ApplicationData.java:112)
	at org.openmicroscopy.shoola.agents.treeviewer.view.TreeViewerFactory.readExternalApplications(TreeViewerFactory.java:477)
	at org.openmicroscopy.shoola.agents.treeviewer.view.TreeViewerFactory.getTreeViewer(TreeViewerFactory.java:427)
	at org.openmicroscopy.shoola.agents.treeviewer.view.TreeViewerFactory.getTreeViewer(TreeViewerFactory.java:193)
	at org.openmicroscopy.shoola.agents.treeviewer.TreeViewerAgent.activate(TreeViewerAgent.java:661)
	at org.openmicroscopy.shoola.env.Container.activateAgents(Container.java:327)
	at org.openmicroscopy.shoola.env.Container.startService(Container.java:360)
	at org.openmicroscopy.shoola.env.init.Initializer.notifyEnd(Initializer.java:197)
	at org.openmicroscopy.shoola.env.Container.runStartupProcedure(Container.java:119)
	at org.openmicroscopy.shoola.env.Container.access$000(Container.java:68)
	at org.openmicroscopy.shoola.env.Container$1.run(Container.java:164)
	at java.lang.Thread.run(Thread.java:748)
Abnormal termination due to an uncaught exception.
java.lang.NoSuchMethodError: com.sun.jna.Native.load(Ljava/lang/String;Ljava/lang/Class;Ljava/util/Map;)Lcom/sun/jna/Library;
	at com.sun.jna.platform.win32.Version.<clinit>(Version.java:36)
	at org.openmicroscopy.shoola.env.data.model.appdata.WindowsApplicationDataExtractor.extractAppData(WindowsApplicationDataExtractor.java:235)
	at org.openmicroscopy.shoola.env.data.model.ApplicationData.<init>(ApplicationData.java:112)
	at org.openmicroscopy.shoola.agents.treeviewer.view.TreeViewerFactory.readExternalApplications(TreeViewerFactory.java:477)
	at org.openmicroscopy.shoola.agents.treeviewer.view.TreeViewerFactory.getTreeViewer(TreeViewerFactory.java:427)
	at org.openmicroscopy.shoola.agents.treeviewer.view.TreeViewerFactory.getTreeViewer(TreeViewerFactory.java:193)
	at org.openmicroscopy.shoola.agents.treeviewer.TreeViewerAgent.activate(TreeViewerAgent.java:661)
	at org.openmicroscopy.shoola.env.Container.activateAgents(Container.java:327)
	at org.openmicroscopy.shoola.env.Container.startService(Container.java:360)
	at org.openmicroscopy.shoola.env.init.Initializer.notifyEnd(Initializer.java:197)
	at org.openmicroscopy.shoola.env.Container.runStartupProcedure(Container.java:119)
	at org.openmicroscopy.shoola.env.Container.access$000(Container.java:68)
	at org.openmicroscopy.shoola.env.Container$1.run(Container.java:164)
	at java.lang.Thread.run(Thread.java:748)
Exception in thread "Initializer"

	at org.openmicroscopy.shoola.env.ui.UserNotifierImpl.showErrorDialog(UserNotifierImpl.java:189)
	at org.openmicroscopy.shoola.env.ui.UserNotifierImpl.notifyError(UserNotifierImpl.java:289)
	at org.openmicroscopy.shoola.env.AbnormalExitHandler.doTermination(AbnormalExitHandler.java:147)
	at org.openmicroscopy.shoola.env.AbnormalExitHandler.terminate(AbnormalExitHandler.java:85)
	at org.openmicroscopy.shoola.env.RootThreadGroup.uncaughtException(RootThreadGroup.java:69)
	at java.lang.Thread.dispatchUncaughtException(Thread.java:1959)

**Comment by dominikl** [comment]
I don't have a `build/distributions` after running `gradle imageJJar`.

**Comment by jburel** [comment]
@dominikl you are correct. You need to do ``gradle build`` to have the ``distributions`` folder

**Comment by rgozim** [comment]
> I don't have a `build/distributions` after running `gradle imageJJar`.

You could also run `gradle imagejDistZip`

**Comment by jburel** [comment]
https://medium.com/@adam_carroll/java-packager-with-jdk11-31b3d620f4a8

**Comment by jburel** [comment]
I built the dmg using 
```
java -version
openjdk version "11.0.2" 2019-01-15
OpenJDK Runtime Environment AdoptOpenJDK (build 11.0.2+9)
OpenJDK 64-Bit Server VM AdoptOpenJDK (build 11.0.2+9, mixed mode)
```
This was installed using ``brew cask install adoptopenjdk``

File: src/main/java/org/openmicroscopy/shoola/util/ui/filechooser/GenericFileChooser.java [hunk 8]
```diff
@@ -69,8 +68,7 @@
 	private void handleFileSelection(File f)
 	{
 		if (box == null || f == null) return;
-		if ((f instanceof ShellFolder && ((ShellFolder) f).isLink()) ||
-				f.getName().endsWith(".lnk")) {
+		if (Files.isSymbolicLink(Paths.get(f.getAbsolutePath()))) {
```
Comment (jburel) [inline src/main/java/org/openmicroscopy/shoola/util/ui/filechooser/GenericFileChooser.java]: why is this change included in this PR?

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [OMERO-insight-push#44](https://web-proxy.openmicroscopy.org/west-ci/job/OMERO-insight-push/44/). See the [console output](https://web-proxy.openmicroscopy.org/west-ci/job/OMERO-insight-push/44/consoleText) for more details.
Possible conflicts:
  - PR #6 jburel 'Remove usage of ShellFolder'
    - src/main/java/org/openmicroscopy/shoola/util/ui/filechooser/GenericFileChooser.java

--conflicts

File: src/main/java/org/openmicroscopy/shoola/util/ui/filechooser/GenericFileChooser.java (hunk 8 above)
Comment (rgozim) [reply src/main/java/org/openmicroscopy/shoola/util/ui/filechooser/GenericFileChooser.java]: I did a git pull instead of a reset hard from my home PC, which I'd used to build the exe I sent you. I can remove it from the PR if you want?

File: build.gradle [hunk 9]
```diff
@@ -30,99 +38,11 @@ dependencies {
...
-                    details.value = baseValue
-                } else {
-                    details.value = details.mergeValue
-                }
-            }
-        }
-    }
-
-}
+    implementation("org.swinglabs.swingx:swingx-core:1.6.5-1")
```
Comment (jburel) [inline build.gradle]: I think we need to be conservative here. Bumping the version of swing will require a full review of the app.

File: buildSrc/src/main/groovy/org/openmicroscopy/InsightBasePlugin.groovy [hunk 10]
```diff
@@ -0,0 +1,114 @@
...
+    private Action<? extends Task> addManifest(String mainClass) {
+        return new Action<Jar>() {
+            @Override
+            void execute(Jar jar) {
+                jar.manifest.attributes(createBasicManifest(mainClass))
+            }
+        }
+    }
+
+    private Map<String, ?> createBasicManifest(String mainClass) {
```
Comment (jburel) [inline buildSrc/src/main/groovy/org/openmicroscopy/InsightBasePlugin.groovy]: can we use the task from the publishing plugin?

File: buildSrc/src/main/groovy/org/openmicroscopy/InsightPlugin.groovy [hunk 11]
```diff
@@ -0,0 +1,106 @@
+package org.openmicroscopy
```
Comment (jburel) [inline buildSrc/src/main/groovy/org/openmicroscopy/InsightPlugin.groovy]: missing header with license

File: buildSrc/src/main/groovy/org/openmicroscopy/DistributePlugin.groovy [hunk 12]
```diff
@@ -0,0 +1,69 @@
+package org.openmicroscopy
```
Comment (jburel) [inline buildSrc/src/main/groovy/org/openmicroscopy/DistributePlugin.groovy]: missing header with license

File: buildSrc/build.gradle [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+        distribute {
+            id = "org.openmicroscopy.distribute"
+            implementationClass = "org.openmicroscopy.DistributePlugin"
+        }
+        publish {
+            id = "org.openmicroscopy.publish"
+            implementationClass = "org.openmicroscopy.PublishPlugin"
+        }
+    }
+}
```
Comment (jburel) [inline buildSrc/build.gradle]: add extra line

File: buildSrc/src/main/groovy/org/openmicroscopy/PublishPlugin.groovy [hunk 14]
```diff
@@ -0,0 +1,58 @@
+package org.openmicroscopy
```
Comment (jburel) [inline buildSrc/src/main/groovy/org/openmicroscopy/PublishPlugin.groovy]: missing header with license

**Comment by jburel** [comment]
@rgozim could you fix the last round of comments?

File: build.gradle (hunk 9 above)
Comment (jburel) [reply build.gradle]: This was already rolled back following testing
https://github.com/ome/omero-insight/pull/1/commits/7675d088e96e1710964784af5101b9ada615bbab

**Comment by jburel** [comment]
testing https://github.com/ome/omero-insight/pull/16/commits/8f45c943fa361a85535d26c1d9274750aeb9ff9a:I tried to run the shell script from various locations and it works

**Comment by jburel** [comment]
Tested fb89cdbd70911040c00ee8e36fe8889a0726ad5e
* run ``gradle build``
* Check under ``build/distributions``. No ``.tar``` created

**Comment by jburel** [comment]
https://github.com/ome/omero-insight/pull/16#issuecomment-484040862 only happens when the app is packaged. Running ``gradle run`` works fine

**Comment by jburel** [comment]
Tested the packaging on Windows7 using https://github.com/ojdkbuild/ojdkbuild/releases/download/1.8.0.201-1/java-1.8.0-openjdk-1.8.0.201-1.b09.ojdkbuild.windows.x86_64.msi (The OpenFX option must be selected to have the javapackager included). I was able to package.

**Comment by jburel** [comment]
@rgozim could you add the package instructions to the README?
Oracle is not required to package

**Comment by rgozim** [comment]
No issues for me running insight, built and installed on Windows 7 with Service Pack 1

**Comment by jburel** [comment]
https://github.com/ome/omero-insight/pull/16#issuecomment-484040862 is due to netcdf pulling an older version of jna

**Comment by jburel** [comment]
Excluding jna coming from netcdf fixes the error noticed https://github.com/ome/omero-insight/pull/16#issuecomment-484040862
The bat file now works as expected.
On both windows and mac it is possible to build the exe/dmg without using oracle java.
README has been updated in #18 
Executable/dmg version will happen in a follow-up PR when the changing in the plugin is working
Merging so we can move towards m2
